### ISSUE TITLE
Dict unpacking and unions lose `Literal` typing

### ISSUE THREAD

**Issue by arjenzorgdoc** [microsoft/pyright#4707]
**Describe the bug**
When concatenating `dict`s with `Literal` keys, I get a type error from pyright.

**To Reproduce**
```python
from typing import Literal

a: dict[Literal["a","b"], str] = {'a': "x"}
b: dict[Literal["a","b"], str] = {'b': "y"}
b = {**a, **b}
b = a | b
```

```
$ npx pyright@1.1.296  test2.py 
No configuration file found.
No pyproject.toml file found.
stubPath /home/arjen/tmp/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.296
/home/arjen/tmp/test2.py
  /home/arjen/tmp/test2.py:5:5 - error: Expression of type "dict[str, str]" cannot be assigned to declared type "dict[Literal['a', 'b'], str]" (reportGeneralTypeIssues)
  /home/arjen/tmp/test2.py:6:5 - error: Expression of type "dict[str, str]" cannot be assigned to declared type "dict[Literal['a', 'b'], str]"
    "dict[str, str]" is incompatible with "dict[Literal['a', 'b'], str]"
      TypeVar "_KT@dict" is invariant
        Type "str" cannot be assigned to type "Literal['a', 'b']"
          "str" cannot be assigned to type "Literal['a']" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations 
Completed in 0.584sec
```

**Expected behavior**
Dict unpacking or unions of dict should return the original type.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.297, which I just published. It will also be included in a future release of pylance.
